### PR TITLE
Extending CSE support

### PR DESCRIPTION
**Pull request by PetarMax**
This PR provides infrastructure and tests in order to extend the support we have for CSE. In particular:

- it adds a number of CSE-related tests into a separate CI job; some are not run because of a bug in the booster that is to be fixed by [this PR](https://github.com/runtimeverification/hs-backend-booster/pull/544).
- To relieve pressure off of the CI, we are running only the CSE versions of the CSE-related tests and commenting out not the non-CSE ones. Eventually, both should be run since they will allow for a precise comparison of the effects of CSE.
- It adds a file with lemmas useful for CSE. These lemmas may be upstreamable to KEVM in subsequent PRs.

### CODE REVIEW

**Comment by ehildenb** [comment]
Make sure to coordinate with @anvacaru , who has some work here: https://github.com/runtimeverification/kontrol/pull/347.

I think it may be useful to focus on internal functions in parallel to his work on automating CSE.

File: src/tests/integration/test-data/foundry/src/cse/lemmas.k [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      requires #rangeUInt(256, X) andBool 0 <=Int Y
+      [simplification, concrete(Y)]
+
+    // Set equality needed for discharging `#Not ( #Exists ( ... )` on `<accessedAccounts>` unification
+    rule { S1:Set #Equals S2:Set |Set SetItem ( X ) } =>
+         { X in S1 } #And
+         ( { S2 #Equals S1 } #Or { S2 #Equals S1 -Set SetItem ( X ) } )
+         [simplification]
+
+endmodule
```
Comment (ehildenb) [inline src/tests/integration/test-data/foundry/src/cse/lemmas.k:28]: These  seem like lemmas we can get upstreamed into KEVM with appropriate tests (the motivating expressions that caused these simplifications to be needed).

Comment (PetarMax) [reply src/tests/integration/test-data/foundry/src/cse/lemmas.k:28]: The first two, yes, the third one is needed until the KCFG-to-KCFG transformation that summarises the behaviour in one-step is done.

**Comment by ehildenb** [comment]
@PetarMax @anvacaru this is becoming quite a huge change, and I'm worried about it stalling out! Is there a subset of the changes that are working well that can be cherry-picked into its own PR?? Then we can get taht merged, and rebase this. Always best to get functionality and tests merged as soon as possible, so everyone is working from the common base and that functionality is guaranteed to be working by CI.

File: src/tests/integration/test-data/foundry-dependency-all [hunk 2]
```diff
@@ -1,3 +1,8 @@
 ArithmeticCallTest.test_double_add(uint256,uint256)
 ArithmeticCallTest.test_double_add_double_sub(uint256,uint256)
 ArithmeticCallTest.test_double_add_sub_external(uint256,uint256,uint256)
+AddConst.applyOp(uint256)
+CSETest.test_add_const(uint256,uint256)
+CSETest.test_identity(uint256,uint256)
+Identity.applyOp(uint256)
+Identity.identity(uint256)
```
Comment (anvacaru) [inline src/tests/integration/test-data/foundry-dependency-all:8]:
```suggestion
AddConst.applyOp(uint256)
ArithmeticCallTest.test_double_add(uint256,uint256)
ArithmeticCallTest.test_double_add_double_sub(uint256,uint256)
ArithmeticCallTest.test_double_add_sub_external(uint256,uint256,uint256)
CSETest.test_add_const(uint256,uint256)
CSETest.test_identity(uint256,uint256)
Identity.applyOp(uint256)
Identity.identity(uint256)
```

File: src/tests/integration/test-data/foundry-dependency-skip [hunk 3]
```diff
@@ -1 +1,5 @@
 ArithmeticCallTest.test_double_add_sub_external(uint256,uint256,uint256)
+AddConst.applyOp(uint256)
+CSETest.test_identity(uint256,uint256)
+CSETest.test_add_const(uint256,uint256)
+Identity.identity(uint256)
```
Comment (anvacaru) [inline src/tests/integration/test-data/foundry-dependency-skip:5]:
```suggestion
AddConst.applyOp(uint256)
ArithmeticCallTest.test_double_add_sub_external(uint256,uint256,uint256)
CSETest.test_add_const(uint256,uint256)
CSETest.test_identity(uint256,uint256)
Identity.identity(uint256)
```

File: src/tests/integration/test_kontrol_cse.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    #     pending=False,
+    #     failing=False,
+    #     failure_info=False,
+    #     counterexample_info=False,
+    #     port=server.port,
+    # )
+    #
+    # assert_or_update_show_output(
+    #     show_res, TEST_DATA_DIR / f'show/{test_id}.no-cse.expected', update=update_expected_output
+    # )
```
Comment (anvacaru) [inline src/tests/integration/test_kontrol_cse.py:137]: I'd say to remove this unused code since it's pretty straightforward.

**Review by anvacaru — APPROVED** [review]

**Comment by PetarMax** [comment]
There is an adjustment that needs to be made, please don't merge this PR until the following commit.

**Pull request merged**